pkg/scheduler/framework/types.go

```diff
@@ -591,6 +591,13 @@ type NodeInfo struct {
 	Generation int64
 }
 
+// NodeInfo implements KMetadata, so for example klog.KObjSlice(nodes) works
+// when nodes is a []*NodeInfo.
+var _ klog.KMetadata = &NodeInfo{}
+
+func (n *NodeInfo) GetName() string      { return n.node.Name }
```

Comment:
should we do `if n != nil && n.node != nil` just in case?

Reply:
I don't know whether NodeInfo may have a nil node pointer, but that definitely looks worth checking. Added.

Pushed, please take another look.